nilearn/input_data/base_masker.py

```diff
@@ -5,75 +5,85 @@
 # License: simplified BSD
 
 import warnings
+import abc
 
 import numpy as np
-import itertools
 
 from sklearn.base import BaseEstimator, TransformerMixin
-from sklearn.externals.joblib import Memory, Parallel, delayed
+from sklearn.externals.joblib import Memory
 
 from .. import masking
 from .. import image
 from .. import signal
 from .. import _utils
 from .._utils.cache_mixin import CacheMixin, cache
-from .._utils.class_inspect import enclosing_scope_name, get_params
-from .._utils.compat import _basestring, izip
-from nilearn._utils.niimg_conversions import (
-    _iter_check_niimg, _check_same_fov)
-
-
-def filter_and_mask(imgs, mask_img_,
-                    parameters,
-                    memory_level=0,
-                    memory=Memory(cachedir=None),
-                    verbose=0,
-                    confounds=None,
-                    copy=True,
-                    sample_mask=None):
+from .._utils.class_inspect import enclosing_scope_name
+from .._utils.compat import _basestring
+
+
+def filter_and_extract(imgs, extraction_function,
+                       parameters,
+                       memory_level=0, memory=Memory(cachedir=None),
+                       verbose=0,
+                       confounds=None,
+                       copy=True):
+    """Extract representative time series using given function.
+
+    Parameters
+    ----------
+    imgs: 3D/4D Niimg-like object
+        Images to be masked. Can be 3-dimensional or 4-dimensional.
+
+    extraction_function: function
+        Function used to extract the time series from 4D data.
+
+    For all other parameters refer to NiftiMasker documentation
+    """
+    # Since the calling class can be any *Nifti*Masker, we look for exact type
+    if verbose > 0:
+        class_name = enclosing_scope_name(stack_level=3)
+
     # If we have a string (filename), we won't need to copy, as
     # there will be no side effect
-
     if isinstance(imgs, _basestring):
         copy = False
 
     if verbose > 0:
-        class_name = enclosing_scope_name(stack_level=2)
-
-    mask_img_ = _utils.check_niimg_3d(mask_img_)
-
+        print("[%s] Loading data from %s" % (
+            class_name,
+            _utils._repr_niimgs(imgs)[:200]))
     imgs = _utils.check_niimg(imgs, atleast_4d=True, ensure_ndim=4)
+
+    sample_mask = parameters.get('sample_mask', None)
     if sample_mask is not None:
         imgs = image.index_img(imgs, sample_mask)
 
-    # Resampling: allows the user to change the affine, the shape or both
-    if verbose > 1:
-        print("[%s] Resampling" % class_name)
-
-    # Check whether resampling is truly necessary. If so, crop mask
-    # as small as possible in order to speed up the process
-    if not _check_same_fov(imgs, mask_img_):
-        # now we can crop
-        mask_img_ = image.crop_img(mask_img_, copy=False)
+    target_shape = parameters.get('target_shape', None)
+    target_affine = parameters.get('target_affine', None)
```

Comment:
You don't need None in this line and the line above, `{}.get('whatever')` returns None.